### PR TITLE
Swift: Add sinks for the GRDB library

### PR DESCRIPTION
**Pull request by atorralba**
This PR adds sinks for the GRDB library affecting the following queries:

* SQLi
* Cleartext storage in local database
* Path injection

### CODE REVIEW

File: swift/ql/test/query-tests/Security/CWE-311/testGRDB.swift [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+func testCommonTableExpression(password: String, harmless: String) {
+    let _ = CommonTableExpression(named: "", sql: "", arguments: [password]) // BAD
+    let _ = CommonTableExpression(named: "", sql: "", arguments: [harmless]) // GOOD
+    let _ = CommonTableExpression(named: "", columns: nil, sql: "", arguments: [password]) // BAD
+    let _ = CommonTableExpression(named: "", columns: nil, sql: "", arguments: [harmless]) // GOOD
+    let _ = CommonTableExpression(recursive: false, named: "", sql: "", arguments: [password]) // BAD
+    let _ = CommonTableExpression(recursive: false, named: "", sql: "", arguments: [harmless]) // GOOD
+    let _ = CommonTableExpression(recursive: false, named: "", columns: nil, sql: "", arguments: [password]) // BAD
+    let _ = CommonTableExpression(recursive: false, named: "", columns: nil, sql: "", arguments: [harmless]) // GOOD
+}
```
Comment (geoffw0) [inline swift/ql/test/query-tests/Security/CWE-311/testGRDB.swift:214]: There's another way for sensitive data to end up in the database via GRDB - as part of a database record, without using SQL.  I think the pattern we'd be looking for is something like:
```
struct MyData: MutablePersistableRecord { // or PersistableRecord
	var data: String
}

...

var myData = MyData(
  data: password
)

OR

myData.data = password
```
This might be best left for another PR.

**Review by geoffw0 — COMMENTED** [review]
This looks great, very thorough coverage!

File: swift/ql/test/query-tests/Security/CWE-311/testGRDB.swift (hunk 1 above)
Comment (atorralba) [reply swift/ql/test/query-tests/Security/CWE-311/testGRDB.swift:214]: > There's another way for sensitive data to end up in the database via GRDB - as part of a database record, without using SQL. 

Good point! And that will probably be the more common case too.

> This might be best left for another PR.

I think so too, this one has enough changes already.

File: swift/ql/test/query-tests/Security/CWE-089/GRDB.swift [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+    let _ = SQLRequest(sql: remoteString, cached: false) // BAD
+    let _ = SQLRequest(sql: localString) // GOOD
+    let _ = SQLRequest(sql: localString, arguments: StatementArguments()) // GOOD
+    let _ = SQLRequest(sql: localString, arguments: StatementArguments(), adapter: nil) // GOOD
+    let _ = SQLRequest(sql: localString, arguments: StatementArguments(), cached: false) // GOOD
+    let _ = SQLRequest(sql: localString, arguments: StatementArguments(), adapter: nil, cached: false) // GOOD
+    let _ = SQLRequest(sql: localString, adapter: nil) // GOOD
+    let _ = SQLRequest(sql: localString, adapter: nil, cached: false) // GOOD
+    let _ = SQLRequest(sql: localString, cached: false) // GOOD
+}
```
[the remark this answers is not in the document]
Comment (atorralba) [reply swift/ql/test/query-tests/Security/CWE-089/GRDB.swift:160]: Good catch, thanks! Added in eb78661c1f3b022f6c5eaa26df860ac6a5b7bdce.

File: swift/ql/test/query-tests/Security/CWE-311/testGRDB.swift (hunk 1 above)
Comment (geoffw0) [reply swift/ql/test/query-tests/Security/CWE-311/testGRDB.swift:214]: I've added a note to the issue.

**Review by geoffw0 — APPROVED** [review]
LGTM.
